src/types/value.rs

```diff
@@ -396,7 +392,16 @@ impl DataValue {
             DataValue::UInt16(v) => v.map(|v| v.encode_fixed_vec()),
             DataValue::UInt32(v) => v.map(|v| v.encode_fixed_vec()),
             DataValue::UInt64(v) => v.map(|v| v.encode_fixed_vec()),
-            DataValue::Utf8(v) => v.clone().map(|v| v.into_bytes()),
+            DataValue::Utf8(v) => v.clone().map(|mut v| {
+                if let Some(LogicalType::Char(len)) = logical_type {
+                    let difference = (len as usize).saturating_sub(v.len());
+
+                    for _ in 0..difference {
+                        v.push(' ')
+                    }
+                }
```

Comment:
maybe use this instead:

```rust
v.push(" ".repeat(difference));
// or
v = format!("{:len$}", v);
```

Reply:
I used `v = format!("{:len$}", v);` This is the first time I saw this usage. cool!